pkgs/top-level/all-packages.nix

```diff
@@ -20931,7 +20931,8 @@ with pkgs;
   ctranslate2 = callPackage ../development/libraries/ctranslate2 rec {
     stdenv = if withCUDA then gcc11Stdenv else pkgs.stdenv;
     withCUDA = pkgs.config.cudaSupport;
-    withCuDNN = pkgs.config.cudaSupport;
+    withCuDNN = withCUDA && (cudaPackages ? cudnn);
```

Comment:
The definition is going to need a `meta.broken = !(withCuda -> cudaPackages ? cudnn)`